session_0/item.rb

```diff
@@ -41,5 +47,15 @@ def discounted?
   # Note: If there are no items for category, stock price for category
   # should be zero.
   def self.stock_price_by_category(items)
+    total = {
```

Comment:
Which one should I implement
```
    total = Hash.new()
    CATEGORIES.each { |key| total[key[0]]=0 }
    items.each do |x|
        total[x.category_id]+=x.current_price * x.quantity
    end
```
or
```
this can be used but the hash it returns will not always have 4(number of CATEGORIES) keys
 total = Hash.new(0)
  items.each do |x|
      total[x.category_id]+=x.current_price * x.quantity
   end
```

Reply:
Which one should I implement
```
    total = Hash.new()
    CATEGORIES.each { |key| total[key[0]]=0 }
    items.each do |x|
        total[x.category_id]+=x.current_price * x.quantity
    end
```
or
```
#this can be used but the hash it returns will not always have 4(number of CATEGORIES) keys
 total = Hash.new(0)
  items.each do |x|
      total[x.category_id]+=x.current_price * x.quantity
   end
```

Reply:
Implement both and see if the tests pass